ocpp/v16/enums.py

```diff
@@ -31,6 +31,18 @@ class Action(str, Enum):
     TriggerMessage = "TriggerMessage"
     UnlockConnector = "UnlockConnector"
     UpdateFirmware = "UpdateFirmware"
+    # Backported actions, Improved security for OCPP 1.6-J specification
+    CertificateSigned = "CertificateSigned"
+    DeleteCertificate = "DeleteCertificate"
+    ExtendedTriggerMessage = "ExtendedTriggerMessage"
+    GetInstalledCertificateIds = "GetInstalledCertificateIds"
+    GetLog = "GetLog"
+    InstallCertificate = "InstallCertificate"
+    LogStatusNotification = "LogStatusNotification"
+    SecurityEventNotification = "SecurityEventNotification"
+    SignCertificate = "SignCertificate"
+    SignedFirmwareStatusNotification = "SignedFirmwareStatusNotification"
+    SignedUpdateFirmware = "UpdateFirmware"
```

Comment:
I guess it was supposed to be SignedUpdateFirmware

Reply:
Also in here: https://github.com/mobilityhouse/ocpp/pull/206/files#diff-8d8918e730a141086aa194d66dabdf5b389795650190b713f7a2e0b29d8ce04dL287

the following fields are missing:
* DownloadScheduled
* DownloadPaused
* InstallRebooting
* InstallScheduled
* InstallVerificationFailed
* InvalidSignature
* SignatureVerified

we also have repeated this fields, so please, if you dont mind could you remove them? @OrangeTux do we know why we have these fields repeated?
https://github.com/mobilityhouse/ocpp/blob/8b055578660c96740df99d4e2614e8157b51929b/ocpp/v16/enums.py#L312

Reply:
This comment is a little unclear for me. Enum values for FirmwareStatusNotificationPayload must be different compared to SignedFirmwareStatus. 

@dataclass
class FirmwareStatusNotificationPayload:
    status: FirmwareStatus

class FirmwareStatus(str, Enum):
    """
    Status of a firmware download as reported in FirmwareStatusNotification.req
    """

    downloaded = "Downloaded"
    download_failed = "DownloadFailed"
    downloading = "Downloading"
    idle = "Idle"
    installation_failed = "InstallationFailed"
    installing = "Installing"
    installed = "Installed"

@dataclass
class SignedFirmwareStatusNotificationPayload:
    status: SignedFirmwareStatus
    request_id: int

----

class SignedFirmwareStatus(str, Enum):
    """
    FirmwareStatusEnumType is used by: SignedFirmwareStatusNotification.req
    """

    certificate_verified = "CertificateVerified"
    downloaded = "Downloaded"
    download_failed = "DownloadFailed"
    downloading = "Downloading"
    download_scheduled = "DownloadScheduled"
    download_paused = "DownloadPaused"
    idle = "Idle"
    installation_failed = "InstallationFailed"
    installing = "Installing"
    installed = "Installed"
    install_rebooting = "InstallRebooting"
    install_scheduled = "InstallScheduled"
    install_verification_failed = "InstallVerificationFailed"
    invalid_signature = "InvalidSignature"
    invalid_certificate = "InvalidCertificate"
    revoked_certificate = "RevokedCertificate"
    signature_verified = "SignatureVerified"

Reply:
I didnt notice that class. However, I must say you are basing your code in a outdated version of the whitepaper. I uploaded here both the spec and the schemas
[OCPP 1.6 security whitepaper edition 2.zip](https://github.com/mobilityhouse/ocpp/files/6435252/OCPP.1.6.security.whitepaper.edition.2.zip)

Reply:
Nevertheless, again the Spec doesnt distinguish between SignedFirmwareStatusEnum and FirmwareStatusEnum, because they share the same core, i.e., adding more options to FirmwareStatusEnum is still backwards compatible with a plain OCPP 1.6J version.

Reply:
You're definitely right. Removed 'InvalidCertificate', 'RevokedCertificate', 'CertificateVerified' enums and added new options to FirmwareStatus and removed SignedFirmwareStatus.